### ISSUE TITLE
iPAS AI 應用規劃師科目3練習 - 2026-08-15

### ISSUE THREAD

**Issue by JohnnyLai92** [JohnnyLai92/ipas#10]
# iPAS AI 應用規劃師｜科目3 機器學習技術與應用 練習（10 題）

本次練習共 **10 題**，涵蓋科目3 四大主題（L231 基礎數學／L232 機器學習與深度學習／L233 建模與參數調校／L234 機器學習治理），題型混合**單選、多選與情境題**，並含 1 題程式判讀。

> ⚠️ 請直接在**本 issue 內文**勾選答案，**不需另外留言或貼答案區**。
> 每題已標示「題型」與「考點」，但**未揭露正確答案**。

## 作答方式

1. 點擊右上角 ✏️ **編輯本 issue 內文**。
2. 把你選擇的選項由 `- [ ]` 改成 `- [x]`（勾起來）。
3. **單選題只勾 1 項；多選題請依題目標示的可勾選數量作答。**
4. 勾選完成、儲存後，將本 issue 的 **label 改為 `ai-analysis`**，即可進入批改與錯題分析。

---

## 第 1 題〔單選〕考點：L23201 機器學習原理／問題類型判斷

某電商想根據顧客的歷史消費紀錄，預測「下個月每位顧客的預期消費金額（連續數值）」。此問題最適合歸類為下列何者？

- [ ] (A) 分類（Classification）
- [ ] (B) 回歸（Regression）
- [x] (C) 分群（Clustering）
- [ ] (D) 降維（Dimensionality Reduction）

---

## 第 2 題〔單選〕考點：L23303 評估指標情境選擇

某醫院導入癌症篩檢模型，最不希望發生的情況是「真正罹病的病患被判定為健康而被漏掉」。在此情境下，應優先關注並提升下列哪一項指標？

- [ ] (A) Precision（精確率）
- [x] (B) Recall（召回率）
- [ ] (C) 整體 Accuracy（準確率）
- [ ] (D) 訓練速度

---

## 第 3 題〔單選〕考點：L23302 Overfitting 判讀

某模型的 Training Accuracy 為 99%，但 Validation Accuracy 只有 72%，兩者差距明顯。最可能發生的問題是？

- [ ] (A) 欠擬合（Underfitting）
- [x] (B) 過擬合（Overfitting）
- [ ] (C) 資料量過多導致收斂太快
- [ ] (D) 學習率設定過低

---

## 第 4 題〔單選〕考點：L23301 Data Leakage

在進行 K-Fold Cross Validation 前，工程師先用**整份資料**（含驗證折）計算 `StandardScaler` 的平均值與標準差再做標準化。此流程最可能造成什麼問題？

- [x] (A) 資料外洩（Data Leakage），驗證分數被高估
- [ ] (B) 模型無法收斂
- [ ] (C) 造成嚴重的欠擬合
- [ ] (D) 沒有任何問題，是正確的標準做法

---

## 第 5 題〔單選〕考點：L23202 演算法特性／可解釋性

某銀行需要一個可以**清楚呈現決策規則、易於向稽核人員解釋**的模型來輔助審核。在下列選項中，通常最符合「高可解釋性且能呈現決策規則」需求的是？

- [ ] (A) 深度神經網路（DNN）
- [x] (B) 決策樹（Decision Tree）
- [ ] (C) 支援向量機（SVM，使用 RBF 核）
- [ ] (D) K-Means

---

## 第 6 題〔單選〕考點：L23203 深度學習應用／Transfer Learning

某工廠僅有 2,000 張已標記的瑕疵產品圖片，且 GPU 資源有限，希望**快速**建立一個影像分類模型。下列策略最適合的是？

- [ ] (A) 從零開始訓練一個超大型深度網路
- [x] (B) 使用 Transfer Learning：以 Pretrained Model 為基礎再 Fine-tuning
- [ ] (C) 改用 K-Means 對圖片做分群
- [ ] (D) 直接以線性回歸預測瑕疵類別

---

## 第 7 題〔多選（可勾選 4 項）〕考點：L23304 過擬合改善方法

下列哪些做法**有助於改善過擬合（Overfitting）**？（請勾選 4 項）

- [x] (A) 增加訓練資料量或使用 Data Augmentation
- [x] (B) 加入 L1／L2 Regularization 或 Dropout
- [ ] (C) 大幅增加網路層數與參數量
- [x] (D) 採用 Early Stopping（提早停止訓練）
- [x] (E) 適度降低模型複雜度

---

## 第 8 題〔多選（可勾選 3 項）〕考點：L23301 資料準備與特徵工程

關於資料前處理與特徵工程，下列敘述**正確**的有哪些？（請勾選 3 項）

- [x] (A) One-Hot Encoding 常用於將「無順序關係」的類別特徵轉換為數值
- [x] (B) 特徵縮放（Normalization／Standardization）的參數應只以訓練集擬合，再套用到測試集
- [x] (C) 對於數值型特徵，處理缺失值可採用以該欄平均數或中位數填補等方式
- [ ] (D) 為避免麻煩，測試集也應該一起參與 `fit` 以取得更準的縮放參數
- [ ] (E) 類別數量極多的欄位一律直接刪除，不需任何編碼

---

## 第 9 題〔多選（可勾選 3 項）〕考點：L23402 演算法偏見與公平性

某貸款審核模型對兩個「條件相近」的族群，卻產生**核准率大幅不同**的結果。下列哪些是**合理且負責任**的處理方向？（請勾選 3 項）

- [x] (A) 檢查訓練資料是否存在偏差（Data Bias／Historical Bias）
- [x] (B) 檢視所使用的特徵是否隱含敏感或代理變數，並評估不同群體的模型結果
- [x] (C) 使用公平性指標評估，必要時重新訓練或調整模型
- [ ] (D) 只看整體 Accuracy，只要準確率夠高就忽略群體差異
- [ ] (E) 直接刪除測試資料以掩蓋差異

---

## 第 10 題〔單選 · 程式判讀〕考點：L23203 Tensor Shape／Linear Layer

閱讀下列 PyTorch 程式碼，`print(y.shape)` 的輸出為何？

```python
import torch
import torch.nn as nn

model = nn.Linear(4, 2)
x = torch.randn(10, 4)
y = model(x)
print(y.shape)
```

- [ ] (A) torch.Size([10, 4])
- [ ] (B) torch.Size([4, 2])
- [x] (C) torch.Size([10, 2])
- [ ] (D) torch.Size([2, 10])
 
### ✅ 完成後檢查清單

- [ ] 單選題每題只勾 1 項
- [ ] 第 7、8、9 題（多選）已依標示數量勾選
- [ ] 已將本 issue 的 **label 改為 `ai-analysis`** 以進入批改

**Comment by JohnnyLai92** [comment]
**Claude Code 練習批改與題型分析**

### 批改結果

| 題號 | 你的作答 | 正確答案 | 判定 |
|:---:|:---:|:---:|:---:|
| 1 | (C) | **(B)** | ❌ |
| 2 | (B) | (B) | ✅ |
| 3 | (B) | (B) | ✅ |
| 4 | (A) | (A) | ✅ |
| 5 | (B) | (B) | ✅ |
| 6 | (B) | (B) | ✅ |
| 7 | (A)(B)(D)(E) | (A)(B)(D)(E) | ✅ |
| 8 | (A)(B)(C) | (A)(B)(C) | ✅ |
| 9 | (A)(B)(C) | (A)(B)(C) | ✅ |
| 10 | (C) | (C) | ✅ |

**總分：9／10**

**逐題解析：**

- **第 1 題 ❌**：預測「每位顧客的預期消費金額（連續數值）」屬於**監督式學習**中的**回歸（Regression）**，因為目標變數是連續型數值。你選的 (C) 分群是非監督式學習，用於「沒有標籤時把相似樣本分組」，與「預測特定連續數值」的任務不符。(A) 分類用於離散類別、(D) 降維用於壓縮特徵維度，皆非本題目的。**正解 (B)**。
- **第 2 題 ✅**：最怕「罹病者被漏判為健康」＝減少 False Negative ＝提升 **Recall（召回率）**。
- **第 3 題 ✅**：訓練高、驗證低且差距大，典型**過擬合**。
- **第 4 題 ✅**：用整份資料（含驗證折）擬合 Scaler，驗證集資訊洩漏進前處理，造成 **Data Leakage** 使分數高估。
- **第 5 題 ✅**：**決策樹**可輸出明確的 if-then 決策規則，可解釋性高。
- **第 6 題 ✅**：小資料＋GPU 有限＋要快 → **Transfer Learning + Fine-tuning**。
- **第 7 題 ✅**：C（大幅增加層數與參數）會**加劇**過擬合，正確排除；A、B、D、E 皆為有效手段。
- **第 8 題 ✅**：D（測試集一起 fit）會造成 Leakage、E（一律刪除）過於武斷，正確排除。
- **第 9 題 ✅**：D（只看整體 Accuracy）、E（刪測試資料掩蓋差異）為不負責任做法，正確排除。
- **第 10 題 ✅**：`nn.Linear(4,2)` 對輸入 `(10,4)` 只改最後一維 → **(10, 2)**。

---

### 題型分析

本次僅錯 **第 1 題**，考點為 **L23201 機器學習原理／問題類型判斷**。

- **共同觀念弱點**：對「任務類型的第一層判斷」尚未內化。判斷關鍵在於**目標變數（label）的型態**：
  - 連續數值 → **回歸**
  - 離散類別（有標籤）→ **分類**
  - 無標籤、找相似群組 → **分群**
  - 壓縮特徵維度 → **降維**
- 本題的陷阱在於題幹提到「顧客歷史紀錄」容易讓人聯想到顧客分群，但**題目要預測的是一個明確的連續數值（消費金額）**，屬監督式回歸，而非非監督式分群。抓住「要預測什麼、那個目標是連續還是離散」即可正確歸類。

其餘題目（評估指標、過擬合判讀與改善、Data Leakage、可解釋性、Transfer Learning、特徵工程、公平性、Tensor Shape）**全數答對**，觀念掌握相當扎實。

---

### 新的解答建議

**第 1 題修正版：**
> **(B) 回歸（Regression）** — 目標變數「預期消費金額」為連續數值，屬監督式學習中的回歸任務。

**讀書建議：**

1. 建立一張「任務類型速判表」（連續值→回歸／有標籤離散→分類／無標籤找群→分群／降特徵→降維），遇到情境題先問自己「目標是什麼、它是連續還是離散、有沒有標籤」。
2. 特別留意情境題的**干擾敘述**（如本題的「顧客紀錄」），不要被應用場景帶偏，回到「要預測的目標型態」判斷。
3. 你在 L232（監督/非監督細節）、L233（建模與調校）、L234（治理與公平性）已表現優秀，可維持現有節奏，僅需針對 L231/L232 的「問題類型定義」再做幾題情境練習鞏固即可。

整體表現優異，繼續保持！ 👍

**Comment by JohnnyLai92** [comment]
已完成錯題分析並輸出到 `docs/ipas-ai-practice-issue-10.md`。

## Claude Code 歸檔摘要
已建立 `docs/ipas-ai-practice-issue-10.md`，未修改其他檔案。

重點摘要：
- **錯題判定**：以 issue 內文勾選（`- [x]`）為主、留言批改為輔，兩者一致，可靠判定**第 1 題**為唯一錯題（作答 (C) 分群 → 正解 (B) 回歸），總分 9／10。
- **第 1 題錯誤分析**：被「顧客紀錄」情境帶偏誤選分群；正確觀念是依**目標變數型態**判斷任務類型——連續數值（消費金額）屬監督式**回歸**，並附上任務類型速判表與考點（監督/非監督、回歸 vs 分類、情境干擾辨識）。
- 另整理其餘 9 題（Recall、Overfitting、Data Leakage、可解釋性、Transfer Learning、正則化、特徵工程、公平性、Tensor Shape）的答對重點供鞏固。
- **後續複習建議**：建速判表、針對 L231/L232 問題類型定義加練情境題、維持既有強項。
